pkg/controller/datavolume/snapshot-clone-controller.go

```diff
@@ -450,7 +450,12 @@ func (r *SnapshotCloneReconciler) createTempHostAssistedSourcePvc(dv *cdiv1.Data
 	return nil
 }
 
-func (r *SnapshotCloneReconciler) getStorageClassCorrespondingToSnapClass(driver string) (string, error) {
+func (r *SnapshotCloneReconciler) getStorageClassCorrespondingToSnapClass(dv *cdiv1.DataVolume, driver string) (string, error) {
```

Comment:
So this only covers the non-CSI path (`initLegacyClone`), I assume you're more interested in the CSI one?

Reply:
The changes I made are for cases where TempHostAssistedSourcePvc is created, i.e., the csi driver is not present,  or the csi driver for the source and target is mismatched. IIRC both are triggered by  initLegacyClone . the first one is my use case

happy to work on any similar cases. Are you referring to the remaining case, which is the csi driver for target pvc and snapshot match? It tries for a direct snapshot to target pvc restore , but it fails because of a csi incompatibility issue ?

Reply:
Oh right, I forgot the issue details. Let me stress again that what you're describing is very unusual today
> the csi driver is not present

> It tries for a direct snapshot to target pvc restore , but it fails because of a csi incompatibility issue ?

Yes, here's a ref to this path https://github.com/kubevirt/containerized-data-importer/blob/dcf7cfba8a0616cd4931881768057a84fdea960b/pkg/controller/clone/planner.go#L835

Reply:
Thanks! I will work on this

Reply:
So just to poke at the setup a little bit again, the provisioner has snapshots (which I thought implies it's CSI) but there's no CSIDriver object?

Reply:
Yes, that's the case. The provisioner is [openebs mayastor](https://github.com/openebs/mayastor) 
The deployment manifest does not contain a `CSIDriver` packaged within

Reply:
wdyt think about also using [pv.spec.CSI](https://github.com/kubernetes/api/blob/v0.32.0/core/v1/types.go#L348) since it is directly populated in when using a csi driver 

 I found this logic being used by [velero](https://github.com/vmware-tanzu/velero/blob/main/pkg/util/kube/utils.go#L238-L241) in a way

Reply:
> Yes, that's the case. The provisioner is [openebs mayastor](https://github.com/openebs/mayastor) The deployment manifest does not contain a `CSIDriver` packaged within

That sounds unintended, how would you then configure things like `fsGroupPolicy`?
https://kubernetes-csi.github.io/docs/support-fsgroup.html

Reply:
> > Yes, that's the case. The provisioner is [openebs mayastor](https://github.com/openebs/mayastor) The deployment manifest does not contain a `CSIDriver` packaged within
> 
> That sounds unintended, how would you then configure things like `fsGroupPolicy`? https://kubernetes-csi.github.io/docs/support-fsgroup.html

These are filled in as parameters for the storage class. I have confirmed with their community that they do not have a CSI driver

Reply:
> > wdyt think about also using [pv.spec.CSI](https://github.com/kubernetes/api/blob/v0.32.0/core/v1/types.go#L348) since it is directly populated in when using a csi driver
> 
> I don't think it's possible since a PV does not exist at the time CDI is interested in this

so its not evaluated on fly for each dv/pvc ?

Reply:
I don't understand the motivation to avoid a CSIDriver object.. AFAIK it's a requirement https://kubernetes-csi.github.io/docs/csi-driver-object.html

Reply:
> I don't understand the motivation to avoid a CSIDriver object.. AFAIK it's a requirement https://kubernetes-csi.github.io/docs/csi-driver-object.html

Yeah I agree it’s a standard and should be supported :disappointed: 
This seems like a case of delayed adoption to me, as it used to work without it